src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java

```diff
@@ -146,7 +147,16 @@ public SearchResponse handle(IndexState indexState, SearchRequest searchRequest)
 
       searchContext.getResponseBuilder().setHitTimeout(searchContext.getCollector().hadTimeout());
 
-      diagnostics.setFirstPassSearchTimeMs(((System.nanoTime() - searchStartTime) / 1000000.0));
+      long firstPassSearchTime = System.nanoTime() - searchStartTime;
+
+      diagnostics.setFirstPassSearchTimeMs((firstPassSearchTime / 1000000.0));
+
+      if (!searchContext.getRescorers().isEmpty()) {
+        for (RescoreTask rescorer : searchContext.getRescorers()) {
+          hits = rescorer.rescore(s.searcher, hits);
+        }
+        diagnostics.setRescoreTimeMs(((System.nanoTime() - firstPassSearchTime) / 1000000.0));
```

Comment:
firstPassSearchTime is the first pass search duration in ns. You will want to add a `long rescoreStartTime = System.nanoTime();` before you start the rescore and use that to compute the rescore time.